tools/shared.py

```diff
@@ -1224,6 +1227,11 @@ def __getattr__(self, attr):
         raise AttributeError
 
     def __setattr__(self, attr, value):
+      for legacy_attr, fixed_values, error_message in self.attrs['REMOVED_SETTINGS']:
```

Comment:
is we move REMOVED_SETTINGS into this python file we can make a map an you can do:

```
if attr in REMOVED_SETTINGS:
  fixed_values, error_message = REMOVED_SETTINGS[attr]
  if value not in fixed_values:
    logger.error('Invalid command line option -s ' + attr + '=' + str(value) + ': ' + error_message)
    sys.exit(1)
```

Also, why not issue a warning an return here.  There no point in actually applying the setting is there?

Or are you trying to handle the case where user JS code is reading these attributes?

Reply:
Applying the settings is needed, since these become options that user code can `#if` against. User code in their own JS library could have had e.g. `#if MEMFS_APPEND_TO_TYPED_ARRAYS` to have a custom block of code that depends on a setting. If we no longer apply these, such user code can silently break without an error being issued.